### PR TITLE
 kernel configs required for eBPF - docs/kernel_config.md

### PR DESCRIPTION
**Pull request by chethanah**
**Background**
- https://github.com/iovisor/bcc/issues/3484 
- List the kernel configs required for eBPF functionalities

**Changes**
- Added docs/kernel_config.md for the required kernel configs 

Kindly review the doc and let us know for any changes or additions

### CODE REVIEW

**Comment by chethanah** [comment]
CC: @KentaTada

File: docs/kernel_config.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Kernel Configuration for eBPF Feature
```
Comment (chenhengqi) [inline docs/kernel_config.md:1]: How about changing `eBPF` to just `BPF` here and below.

Comment (chenhengqi) [reply docs/kernel_config.md:1]: Feature -> Features.

File: docs/kernel_config.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+| | CONFIG_BPFILTER_UMH | This builds bpfilter kernel module with embedded user mode helper |
+| | CONFIG_NET_CLS_BPF | BPF-based classifier - to classify packets based on programmable BPF (JIT'ed) filters as an alternative to ematches |
+| | CONFIG_NET_ACT_BPF | Execute BPF code on packets. The BPF code will decide if the packet should be dropped or not |
+| | CONFIG_BPF_STREAM_PARSER | Enable this to allow a TCP stream parser to be used with BPF_MAP_TYPE_SOCKMAP |
+| | CONFIG_LWTUNNEL_BPF | Allow to run BPF programs as a nexthop action following a route lookup for incoming and outgoing packets |
+| | CONFIG_NETFILTER_XT_MATCH_BPF | BPF matching applies a linux socket filter to each packet and accepts those for which the filter returns non-zero |
+| | CONFIG_IPV6_SEG6_BPF | To support  BPF seg6local hook. bpf: Add IPv6 Segment Routing helpersy. [Reference](https://github.com/torvalds/linux/commit/fe94cc290f535709d3c5ebd1e472dfd0aec7ee7) |
+| **kprobes** | CONFIG_KPROBE_EVENTS | This allows the user to add tracing events (similar to tracepoints) on the fly via the ftrace interface |
+|  | CONFIG_KPROBES | Enable kprobes-based dynamic events |
```
Comment (chenhengqi) [inline docs/kernel_config.md:33]: IIUC, BPF relies on `CONFIG_KPROBES_ON_FTRACE` not just `CONFIG_KPROBES`.

Comment (chethanah) [reply docs/kernel_config.md:33]: Updated `CONFIG_KPROBES_ON_FTRACE` under kprobes section

**Comment by chethanah** [comment]
Hello @chenhengqi , I've updated based on review. Please verify.

**Review by chenhengqi — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**